xlators/storage/posix/src/posix-aio.c

```diff
@@ -368,9 +366,14 @@ posix_aio_writev(call_frame_t *frame, xlator_t *this, fd_t *fd,
         goto err;
     }
 
+    direct = DIRECT_ALIGNED(offset);
+    for (i = 0; i < count; i++)
+        direct = (direct && DIRECT_ALIGNED(iov[i].iov_base) &&
+                  DIRECT_ALIGNED(iov[i].iov_len));
```

Comment:
Probably the compiler will already optimize this to quit the loop once `direct` becomes false, but I think it's better to make it explicit, like this:

```suggestion
    direct = DIRECT_ALIGNED(offset);
    i = 0;
    while (direct && (i < count)) {
        direct = DIRECT_ALIGNED(iov[i].iov_base) &&
                 DIRECT_ALIGNED(iov[i].iov_len);
        i++;
    }
```